lib/src/Pos/SafeCopy.hs

```diff
@@ -362,8 +362,16 @@ instance SafeCopy (AsBinary a) where
     putCopy = contain . safePut . getAsBinary
 
 instance (Typeable a, Bi a) => SafeCopy (WithHash a) where
-    putCopy = putCopyBi
     getCopy = getCopyBi
+    putCopy = putCopyBi
+
+instance SafeCopy HeavyDlgIndex where
+    getCopy = contain $ HeavyDlgIndex <$> safeGet
+    putCopy x = contain $ safePut $ getHeavyDlgIndex x
+
+instance SafeCopy LightDlgIndices where
+    getCopy = contain $ LightDlgIndices <$> safeGet
+    putCopy x = contain $ safePut $ getLightDlgIndices x
```

Comment:
Why did you write these instances manually instead of generating them (as it's done for 90% types)?

Reply:
I am not really sure what exactly will `deriveSafeCopySimple` generate.  Some other wrappers (that doesn't have their own binary representation) are also implemented in the similar way.

Example:
```
instance SafeCopy (Signature a) where
    putCopy (Signature sig) = contain $ safePut sig
    getCopy = contain $ Signature <$> safeGet
```